### DISCUSSION TITLE
autoscaling cluster in multiple locations

### DISCUSSION THREAD

**Discussion by codeagencybe** [kube-hetzner/terraform-hcloud-kube-hetzner#379] [Q&A] [answered]
Hello @mysticaltech 

I have searched in the issues and discusses but I could not find a clear answer if this already been discussed or not but... 

How does this autoscaling work for agent nodes in multiple locations as the example included shows we need to specify a location for it? 
Does this mean I need to copy the autoscaling block for each location like below example? 
Or does 1 block work for all locations automatically? 

I have agent/worker nodes (CPX51)  and my control planes (CPX31) in all 3 locations. 
So how do I address autocluster scaling for my worker nodes in all their locations? 
The explainer in the kubetf file is targetting server types of the control plane servers?? I was expecting that the worker nodes would scale, not the control planes? 

If my example below is wrong, could somebody perhaps show a correct working example for above scenario so I can try it? 
Thanks!

  ```
Cluster Autoscaler
  Providing at least one map for the array enables the cluster autoscaler feature, default is disabled
  autoscaler_nodepools = [
    {
      name        = "autoscaler"
      server_type = "cpx31" # must be same or better than the control_plane server type (regarding disk size)!
      location    = "fsn1"
      min_nodes   = 0
      max_nodes   = 5
    },
    {
      name        = "autoscaler"
      server_type = "cpx31" # must be same or better than the control_plane server type (regarding disk size)!
      location    = "nbg1"
      min_nodes   = 0
      max_nodes   = 5
    },
    {
      name        = "autoscaler"
      server_type = "cpx31" # must be same or better than the control_plane server type (regarding disk size)!
      location    = "hel1"
      min_nodes   = 0
      max_nodes   = 5
    }



  ]


```

**Comment by ifeulner** [comment, marked as answer]
You need separate pools, yes. 
The control_plane is just used to take a snapshot from, agents are getting scaled.
